selenium/src/test/java/com/deque/html/axecore/selenium/AllowedOriginsTest.java

```diff
@@ -0,0 +1,93 @@
+package com.deque.html.axecore.selenium;
+
+import static org.junit.Assert.*;
+
+import com.deque.html.axecore.providers.EmbeddedResourceAxeProvider;
+import com.deque.html.axecore.providers.StringAxeScriptProvider;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Scanner;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+
+public class AllowedOriginsTest {
+
+  private WebDriver webDriver;
+  private String axeSource;
+  private String axeForceLegacy;
+
+  private String addr() {
+    return "http://localhost:8001";
+  }
+
+  private String downloadFromURL(String url) throws Exception {
+    // https://stackoverflow.com/a/13632114
+    try (InputStream stream = new URL(url).openStream()) {
+      return new Scanner(stream, "UTF-8").useDelimiter("\\A").next();
+    }
+  }
+
+  private Object getAllowedOrigins() {
+    JavascriptExecutor javascriptExecutor = (JavascriptExecutor) webDriver;
+    return javascriptExecutor.executeScript("return axe._audit.allowedOrigins");
+  }
+
+  @Before
+  public void setup() throws Exception {
+    webDriver = new ChromeDriver(new ChromeOptions().setHeadless(true));
+    axeSource = new EmbeddedResourceAxeProvider().getScript();
+    axeForceLegacy = downloadFromURL(addr() + "/axe-force-legacy.js");
+  }
+
+  @After
+  public void teardown() {
+    webDriver.quit();
+  }
+
+  @Test
+  public void shouldNotSetWhenRunningRunPartialAndNotLegacyMode() {
+    webDriver.get(addr() + "/index.html");
+    new AxeBuilder().analyze(webDriver);
+    ArrayList<?> allowedOrigins = (ArrayList<?>) getAllowedOrigins();
+
+    assertEquals(allowedOrigins.get(0), addr());
+  }
+
+  @Test
+  public void shouldNotSetWhenRunningRunPartialAndLegacyMode() {
+    webDriver.get(addr() + "/index.html");
+    new AxeBuilder().setLegacyMode(true).analyze(webDriver);
+    ArrayList<?> allowedOrigins = (ArrayList<?>) getAllowedOrigins();
+
+    assertEquals(allowedOrigins.get(0), addr());
+  }
+
+  @Test
+  public void shouldNotSetWhenRunningLegacySourceAndLegacyMode() {
+    webDriver.get(addr() + "/index.html");
+    new AxeBuilder()
+        .setLegacyMode(true)
+        .setAxeScriptProvider(new StringAxeScriptProvider(axeSource + axeForceLegacy))
```

Comment:
Should use legacy source (and then you don't need the force legacy script appended)